app/uk/gov/hmrc/gform/obligation/ObligationService.scala

```diff
@@ -17,30 +17,123 @@
 package uk.gov.hmrc.gform.obligation
 import java.text.SimpleDateFormat
 
+import cats.data.NonEmptyList
+import cats.kernel.Eq
+import cats.implicits._
+import uk.gov.hmrc.gform.auth.AuthService
+import uk.gov.hmrc.gform.auth.models.MaterialisedRetrievals
+import uk.gov.hmrc.gform.controllers.helpers.FormDataHelpers
 import uk.gov.hmrc.gform.gformbackend.GformConnector
 import uk.gov.hmrc.gform.sharedmodel._
+import uk.gov.hmrc.gform.sharedmodel.form.{Form, FormId, UserData}
 import uk.gov.hmrc.gform.sharedmodel.formtemplate._
 import uk.gov.hmrc.http.HeaderCarrier
-import scala.concurrent.ExecutionContext
+
+import scala.concurrent.{ExecutionContext, Future}
 
 class ObligationService(gformConnector: GformConnector) {
 
   val stringToDate = new SimpleDateFormat("yyyy-MM-dd")
 
-  def lookupObligationsMultiple(formTemplate: FormTemplate)(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
+  private def lookupObligationsMultiple(
+    formTemplate: FormTemplate,
+    authService: AuthService,
+    retrievals: MaterialisedRetrievals,
+    form: Form,
+    hmrcTaxPeriodIdentifiers: NonEmptyList[HmrcTaxPeriod])(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
```

Comment:
Wait, what are we doing here actually? It just occurred to me, but this looks like as contradioction of what I intended to achieve by including 

```
Given the formtemplate has more than 1 hmrcTaxPeriodSelector component 
when parsing this formtemplate 
then reject with explanation that only 1 hmrcTaxPeriodSelector is allowed
```

in ACs in the first place.

Having not following rule this we are only making our life more complicated.

`hmrcTaxPeriodIdentifiers: NonEmptyList[HmrcTaxPeriod]` should be just `hmrcTaxPeriodIdentifier: HmrcTaxPeriod`. Or do I miss something?